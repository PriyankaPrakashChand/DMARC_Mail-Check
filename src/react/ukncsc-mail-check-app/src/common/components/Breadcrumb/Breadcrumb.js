import React from 'react';
import { Link } from 'react-router-dom';
import { Breadcrumb } from 'semantic-ui-react';

export default ({ children }) => (
  <div>
    <Breadcrumb>
      <Breadcrumb.Section>
        <Link to="/">Home</Link>
      </Breadcrumb.Section>
      {children}
    </Breadcrumb>
  </div>
);
