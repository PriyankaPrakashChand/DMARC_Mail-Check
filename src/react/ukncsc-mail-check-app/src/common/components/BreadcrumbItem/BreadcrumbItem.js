import React from 'react';
import { Breadcrumb } from 'semantic-ui-react';
import { Link } from 'react-router-dom';

const BreadcrumbItem = ({ link, children, ...props }) => (
  <React.Fragment>
    <Breadcrumb.Divider icon="right angle" />
    <Breadcrumb.Section {...props}>
      {link ? <Link to={link}>{children}</Link> : children}
    </Breadcrumb.Section>
  </React.Fragment>
);

export default BreadcrumbItem;
