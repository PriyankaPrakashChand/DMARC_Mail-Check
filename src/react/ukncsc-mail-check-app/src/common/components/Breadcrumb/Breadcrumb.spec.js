import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from 'react-testing-library';
import Breadcrumb from './Breadcrumb';
import BreadcrumbItem from '../BreadcrumbItem';

describe('Breadcrumb', () => {
  let container;

  test('it should match the snapshot', () => {
    ({ container } = render(
      <MemoryRouter>
        <Breadcrumb>
          <BreadcrumbItem>ncsc.gov.uk</BreadcrumbItem>
        </Breadcrumb>
      </MemoryRouter>
    ));

    expect(container).toMatchSnapshot();
  });
});
