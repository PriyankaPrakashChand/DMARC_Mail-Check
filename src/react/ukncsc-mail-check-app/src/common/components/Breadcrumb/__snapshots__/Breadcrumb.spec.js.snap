// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb it should match the snapshot 1`] = `
<div>
  <div>
    <div
      class="ui breadcrumb"
    >
      <div
        class="section"
      >
        <a
          href="/"
        >
          Home
        </a>
      </div>
      <i
        aria-hidden="true"
        class="right angle icon divider"
      />
      <div
        class="section"
      >
        ncsc.gov.uk
      </div>
    </div>
  </div>
</div>
`;
